src/EditorFeatures/Core/Extensibility/SignatureHelp/SignatureHelpItems.cs

```diff
@@ -72,7 +74,24 @@ internal class SignatureHelpItems
                 throw new ArgumentException($"{nameof(argumentCount)} < {nameof(argumentIndex)}. {argumentCount} < {argumentIndex}", nameof(argumentIndex));
             }
 
-            this.Items = items;
+            // Adjust the `selectedItem` index if duplicates are able to be removed.
+            var distinctItems = items.Distinct().ToList();
+            if (selectedItem.HasValue && items.Count != distinctItems.Count)
+            {
+                // `selectedItem` index has already been determined to be valid, it now needs to be adjusted to account for duplicates being removed.
+                // E.g.,
+                //   items = {A, A, B, B, C, D}
+                //   selectedItem = 4 (index for item C)
+                // ergo
+                //   distinctItems = {A, B, C, D}
+                //   actualItem = C
+                //   selectedItem = 2 (index for item C)
+                var actualItem = items[selectedItem.Value];
+                selectedItem = distinctItems.IndexOf(actualItem);
+                Debug.Assert(selectedItem.Value >= 0, "actual item was not part of the final list");
```

Comment:
It's seems like it is completely reasonable for the `.Distinct` to drop this item, and not the other item which is identical.  What prevents that?